metadata-ingestion/src/datahub/sql_parsing/tool_meta_extractor.py

```diff
@@ -42,14 +55,45 @@ class ToolMetaExtractor:
     by warehouse query logs.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, looker_user_mapping: Optional[Dict[str, str]] = None) -> None:
         self.report = ToolMetaExtractorReport()
         self.known_tool_extractors: List[Tuple[str, Callable[[QueryLog], bool]]] = [
             (
                 "mode",
                 self._extract_mode_query,
-            )
+            ),
+            (
+                "looker",
+                self._extract_looker_query,
+            ),
         ]
+        # maps user id (as string) to email address
+        self.looker_user_mapping = looker_user_mapping
+
+    @classmethod
+    def create(
+        cls,
+        config: ToolMetaExtractorConfig,
+        graph: Optional[DataHubGraph] = None,
+    ) -> "ToolMetaExtractor":
+        looker_user_mapping: Dict[str, str] = {}
+        if graph and config.looker_platform_details:
+            resource_urn = generate_user_id_mapping_resource_urn(
+                "looker",
+                config.looker_platform_details.platform_instance,
+                config.looker_platform_details.env,
+            )
+            aspect = graph.get_aspect_v2(
+                entity_urn=resource_urn,
+                aspect_type=PlatformResourceInfoClass,
+                aspect="platformResourceInfo",
+            )
+            if aspect and aspect.value:
+                with contextlib.suppress(ValueError, AssertionError):
+                    looker_user_mapping = from_serialized_value(aspect.value)
+                    assert isinstance(looker_user_mapping, dict)
```

Comment:
can we add some reporting in in looker source about how many users were added to platform resource ? It would also help to have some indication in tool meta extractor report about number of looker users were found in mapping vs not.